feat(replay): session replay updates to add AI summary for web (#14794)
getsentry/sentry-docs: docs/product/explore/session-replay/replay-details.mdx

</Alert>

### Timeline

The timeline section at the bottom of the **Replay Details** page illustrates where significant events (such as errors, user interactions, and more) happen over the course of the replay. This allows users to easily scrub to key events by dragging across the timeline. It also visually conveys the amount of time that took place between events and has a zoom functionality, so you can easily zoom in to distinguish between events that happen close together.

## Data Tabs

### Console

A list of debugging messages that don't belong in the breadcrumbs list will appear here. For web, this includes `console.log` statements and browser-generated messages to the developer. For React Native, custom `console.log` will appear here, and in Android, logs from [Logcat](/platforms/android/integrations/logcat/) and [Timber](/platforms/android/integrations/timber/) are also supported.

### Network

> A list of all network requests that were initiated by the app while the replay recording was active. As the video plays, there is a visual indicator that tracks through the table of network requests, highlighting which requests happened prior to or next to this point in the video. You can also click the timestamp on the far right of each request to bring yourself to that point in the replay player.

Reviewer: ```suggestion
This tab contains a list of all network requests that were initiated by the app while the replay recording was active. As the video plays, there is a visual indicator that tracks through the table of network requests, highlighting which requests happened prior to or next to this point in the video. You can also click the timestamp on the far right of each request to bring yourself to that point in the replay player.
```

</Alert>

### AI-Powered Replay Summary [Web Only]

<Alert>
This feature is currently in Beta, which means that this feature is still in-progress and may have bugs. We recognize the irony.
</Alert>

Sentry now provides **AI-powered replay summaries** that automatically analyze what happened during a user session and give you a short play-by-play to help debug issues faster. Rather than watch an entire replay, you can now get a quick overview of what happened and focus on the specific interactions that led to the issue. This tab includes:

- **AI Summary:** An LLM-powered description of the user's journey
- **Chapters:** Each chapter summarizes related sets of breadcrumbs, network requests, and console logs, ordered by time range. Click a chapter to expand its contents and see the underlying events
- **Red Highlighted Chapters:** These mark chapters containing errors
- **Pink Highlighted Chapters:** These mark chapters containing feedback

### Breadcrumbs

The replay breadcrumbs show when key user interactions took place. Breadcrumbs are synced with the replay player and will auto-scroll as the video plays.

**Both web and mobile replays include:**

> - **Web Vitals:** and an overall performance score [Learn more](/product/insights/web-vitals/)

Author: Ohh, I thought your last message said that web vitals were on both. I was so confused. I'll remove it and move it back under web.

**Web-specific breadcrumbs include:**

> - **User Clicks:** including rage and dead clicks

Reviewer 2: ```suggestion
- **User Clicks:** including [rage clicks](/product/issues/issue-details/replay-issues/rage-clicks/) and [dead clicks](/product/issues/issue-details/replay-issues/rage-clicks/#detection-criteria)
```

- **Navigations and Page Loads:** [Learn more about spans](/product/sentry-basics/distributed-tracing/)
- **Custom Breadcrumbs:** Learn more about configuring [custom breadcrumbs](/product/sentry-basics/integrate-backend/configuration-options/#breadcrumbs)

**Mobile-specific breadcrumbs include:**
- Backgrounding and foregrounding
- Network connectivity changes
- Battery usage

<Alert>

> Some [breadcrumb types](/product/issues/issue-details/breadcrumbs/) visible in **Issue Details** are not 1:1 to the replay breadcrumbs list. The trail of events typically seen in the Issue Details page are now displayed in the Console and Network components of the **Replay Details** page.

Reviewer 2: ```suggestion
Some [breadcrumb types](/product/issues/issue-details/breadcrumbs/) visible in **Issue Details** are not a one-for-one comparison to the replay breadcrumbs list. The trail of events typically seen in the Issue Details page are now displayed in the Console and Network components of the **Replay Details** page.
```

Opinionated, but felt like it was clearer to spell this out.

<Alert>

Due to transaction sampling, this view may be missing traces.

</Alert>

### Memory [Web Only]

The view shows a heap size chart displaying the total amount of memory being used by JavaScript objects.

<Alert>

This view is only available when the replay was recorded on a Chromium-based browser.

</Alert>

### Tags

A complete list of built-in fields and custom tags associated with a replay, such as operating system version and name, device specs, release, and user details.

![Tags tab on Replay Details.](./img/replay-details-tags.png)

## Replay Actions

### Share Replays

Share a replay at a specific timestamp with a Sentry member by clicking the "Share" button in the top-right corner of the **Replay Details** page. This allows you to share replays at the point of a critical event or user flow with other Sentry members in your organization.

![Highlight the button to share a replay](./img/replay-details-share.png)

### Delete Replays

Delete an individual replay by clicking the "Delete" button in the top-right corner of the **Replay Details** page. You cannot delete replays that are in progress.

<Alert>

> Deleting replays does not affect your quota.

Reviewer 2: For the sake of avoiding alert fatigue, I feel like we can add this as a tag to the paragraph above, "..in progress. Please note, deleting replays does not . . ."
